### DISCUSSION TITLE
Closure rules have been removed

### DISCUSSION THREAD

**Discussion by gramic** [rules-proto-grpc/rules_proto_grpc#118] [Q&A] [answered]
I am a heavy user of the closure rules in many applications, but just a light one compared to other companies.

1. What is the problem that made you deprecate those rules?
2. With the new js rules, I should use [tsickle](https://github.com/angular/tsickle)? Do you have some example?

**Comment by aaliddell** [comment, marked as answer]
The main reason for the change was that I was running into substantial issues with the closure compiler, for which no solution was documented and I didn't have the bandwidth to support. See for example [this build](https://buildkite.com/bazel/rules-proto-grpc-rules-proto-grpc/builds/832#bf9d0928-2935-4ef0-8080-d141cda99a0b); here Closure compiler was failing with a rather obtuse `java.lang.RuntimeException: INTERNAL COMPILER ERROR` after a `NullPointerException` in `com.google.javascript.jscomp.CheckJSDoc.checkJsDocInBlockComments`. Having dug into the relevant code within the compiler itself, I was none the wiser about why it was tripping up on these files. Something was returning a null pointer after apparently reading the comments in the generated source files and the receiving code doesn't consider that possibility.

I'm completely open to reintroducing the Closure rules if there's continued demand, but as it stands I couldn't see an obvious step to working around the above compiler error.

**Reply by aaliddell** [reply]
If I make a branch that restores closure and isolates that error, would you be in a position to help debug what is causing it?

**Reply by gramic** [reply]
My knowledge of the compiler and java is very limited, so I think that I am not in a good position to help you debug it.

Anyway, I doubt that there will be continued demand for those rules and your decision to drop it is somewhat smart, despite my interest in fixing it. Using tsickle and typescript seems the future direction.
